File: server/python/app/infrastructure/mappers/payment_details/discount_mapper.py
from typing import Optional
from app.application.domain.payments.payment_details.discount import Discount
from onlinepayments.sdk.domain.discount import Discount as DiscountSdk

class DiscountMapper:

    @staticmethod
    def map_from_sdk_response(response: Optional[DiscountSdk]) -> Optional[Discount]:
        if response is None:
            return None

        dto = Discount()
        dto.amount = response.amount

        return dto

File: server/ruby/app/infrastructure/mappers/payment_details/discount_mapper.rb
module Infrastructure
  module Mappers
    module PaymentDetails
      class DiscountMapper
        def self.from_sdk_response(response)
          return nil if response.nil?

          dto = Business::Domain::Payments::PaymentDetails::Discount.new
          dto.amount = response.amount

          dto
        end
      end
    end
  end
end